source/SIL.AppBuilder.Portal.Frontend/src/data/containers/resources/group/with-data-actions.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from 'react';
+import * as toast from '@lib/toast';
+import { compose } from 'recompose';
+import { withData as withOrbit, WithDataProps } from 'react-orbitjs';
+
+import { GroupAttributes, GroupResource } from '@data/models/group';
+import { defaultOptions } from '@data';
+import { requireProps } from '@lib/debug';
+import { OrganizationResource } from '@data';
+
+export interface IProvidedProps {
+  createRecord: (attrs: GroupAttributes) => any;
+  removeRecord: (group: GroupResource) => any;
+}
+
+interface IOwnProps {
+  organization: OrganizationResource;
+}
+
+type IProps =
+  & IOwnProps
+  & WithDataProps;
+
+export function withDataActions(InnerComponent) {
+
+  class DataActionWrapper extends React.Component<IProps> {
+
+    createRecord = async (payload: GroupAttributes) => {
+
+      const { dataStore, organization } = this.props;
+
+      try {
+        await dataStore.update(
```

Comment:
any reason not to use `update` or `create` from `@data`?

Reply:
Yeah I had issues trying to reRender the group list. The list reRender didn't wait for create or update method to finish even though I put async everywhere.

Reply:
but, update / create doesn't have anything to do with async. :-\
async should be used regardless :_\